tensilelite/Tensile/Components/GlobalWriteBatch.py

```diff
@@ -1125,68 +1127,70 @@ def _emitNonatomicAdd(self, module: Module):
 
             # Generate single f32 code if edge is detected.
             if ((vi + 1) == self.gwvw) and ((self.gwvw % 2) == 1):
-              scaleAVecModule.add(VCmpGtU32(dst=sgpr("AddressScaleA", self.parentWriter.states.laneSGPRCount), src0=sgpr("SrdScaleA+2"), src1=0, comment=" == 0 ?"))
-              scaleAVecModule.add(VCndMaskB32(
-                dst=vgpr(inputScaleAVecVgpr), \
-                src1=vgpr(inputScaleAVecVgpr), \
-                src0=1.0, \
-                src2=sgpr("AddressScaleA", self.parentWriter.states.laneSGPRCount), \
-                comment="1. mul 1 if 0"))
+              if not isAChecked:
+                scaleAVecModule.add(VCmpGtU32(dst=sgpr("AddressScaleA", self.parentWriter.states.laneSGPRCount), src0=sgpr("SrdScaleA+2"), src1=0, comment=" == 0 ?"))
+                for vi2 in range(0, self.gwvw):
+                  scaleAVecModule.add(VCndMaskB32(
+                    dst=vgpr(dataScaleAVec + vi2), \
+                    src1=vgpr(dataScaleAVec + vi2), \
+                    src0=1.0, \
+                    src2=sgpr("AddressScaleA", self.parentWriter.states.laneSGPRCount), \
+                    comment="1. mul 1 if 0"))
+                isAChecked = True
```

Comment:
Looks like this isAChecked code can be moved outside for loop?
It will be generated only once?

Reply:
I'll add to my refactor TODO list :)